Expose rdkafka stats to producers and consumers

omnibs opened the pull request.

stoeffel reviewed (approved):
LGTM!

stoeffel committed:
Add stats callback to kafka workers

stoeffel committed:
Add stats callback to kafka

stoeffel committed:
Add type for stats

stoeffel committed:
Decode stats in consumer callback

stoeffel committed:
Decode stats

stoeffel committed:
Expose stats module

omnibs committed:
Let us debug these things

omnibs committed:
Tell kafka to report back with stats every 1s

omnibs committed:
Crash on debug failures

caiquefigueiredo committed:
Show things

caiquefigueiredo committed:
WIP expose dict metrics

stoeffel committed:
Fix test helpers

stoeffel committed:
Add function to decode json into a flat dict

stoeffel committed:
Decode stats into flat dict

zwilias committed:
Keep track of the Path, so not really a flat dict

zwilias committed:
Add pathToText helper

zwilias committed:
Support top-level values

stoeffel committed:
Less casing

stoeffel committed:
Simpler function (maybe slightly slower). Does it matter?

stoeffel committed:
More cleanup

zwilias committed:
Reexpose Path and pathToText

stoeffel committed:
Add topLevel metric types

stoeffel committed:
Attach metric type info to values

stoeffel committed:
Add show instance to metric

stoeffel committed:
Special case `req`

stoeffel committed:
Cleanup

stoeffel committed:
Add req to known metrics

zwilias commented on nri-kafka/src/Kafka/Stats.hs:
```diff
@@ -41,16 +41,27 @@ toMetric path value =
-            _ -> Err ("Metric type mismatch: " ++ pathToText path)
+        Nothing ->
+          -- Get second to last segment of path for `req` case.
+          case List.tail (List.reverse path) of
```
This is still just the last, isn't it? 🤔

zwilias replied:
No it isn't, whoops. Nevermind me :D

omnibs committed:
There's a typo in rdkafka

zwilias committed:
Remove pathToText

It can be moved into hqe, where we actually need a less general
implementation to play nice with prometheus

stoeffel committed:
Formatting

stoeffel committed:
Add todo

stoeffel committed:
Set stats interval to consumer

stoeffel committed:
Add settings for statistics.interval.ms

stoeffel committed:
Use andMap

stoeffel committed:
Format args

stoeffel committed:
Update changelog
